snippets/Google.Storage.V1.Snippets/StorageClientSnippets.cs

```diff
@@ -27,30 +27,73 @@ namespace Google.Storage.V1.Snippets
     [Collection(nameof(StorageSnippetFixture))]
     public class StorageClientSnippets
     {
-        private readonly StorageSnippetFixture fixture;
+        private readonly StorageSnippetFixture _fixture;
 
         public StorageClientSnippets(StorageSnippetFixture fixture)
         {
-            this.fixture = fixture;
+            this._fixture = fixture;
         }
 
         [Fact]
-        void ListBuckets()
+        public void Overview()
         {
-            var projectId = fixture.ProjectId;
+            var projectId = _fixture.ProjectId;
+
+            // <*>
+            var client = StorageClient.Create();
+
+            // Create a bucket
+            var bucketName = Guid.NewGuid().ToString(); // must be globally unique
+            var bucket = client.Service.Buckets.Insert(new Bucket { Name = bucketName }, projectId).Execute();
+
+            // Upload some files
+            var content = Encoding.UTF8.GetBytes("hello, world");
+            var obj1 = client.UploadObject(bucketName, "file1.txt", "text/plain", new MemoryStream(content));
+            var obj2 = client.UploadObject(bucketName, "folder1/file2.txt", "text/plain", new MemoryStream(content));
+
+            // List objects
+            foreach (var obj in client.ListObjects(bucketName, ""))
+            {
+                Console.WriteLine(obj.Name);
+            }
+
+            // Download file
+            using (var stream = File.OpenWrite(@".\file1.txt"))
+            {
+                client.DownloadObject(bucketName, "file1.txt", stream);
+            }
+            // </*>
+
+            Assert.Equal(content, File.ReadAllBytes(@".\file1.txt"));
+            Assert.Contains(client.ListObjects(bucketName, ""), o => o.Name == "file1.txt");
+            Assert.Contains(client.ListObjects(bucketName, ""), o => o.Name == "folder1/file2.txt");
+            Assert.Contains(client.ListBuckets(projectId), b => b.Name == bucketName);
+
+            File.Delete(@".\file1.txt");
+            client.DeleteObject(bucketName, "file1.txt");
```

Comment:
Registering buckets, objects and files for clean-up by the fixture would be useful for the case where the assertions fail, as you say. Good idea.